lib/libmbedtls/core/tee_lmd_provider.c

```diff
@@ -1376,26 +1379,129 @@ TEE_Result crypto_acipher_dsa_verify(uint32_t algo __unused,
 #endif /* CFG_CRYPTO_DSA */
 
 #if defined(CFG_CRYPTO_DH)
-TEE_Result crypto_acipher_alloc_dh_keypair(struct dh_keypair *s __unused,
-					   size_t key_size_bits __unused)
+
+TEE_Result crypto_acipher_alloc_dh_keypair(struct dh_keypair *s,
+					   size_t key_size_bits)
 {
-	return TEE_ERROR_NOT_IMPLEMENTED;
+	memset(s, 0, sizeof(*s));
+	s->g = crypto_bignum_allocate(key_size_bits);
+	if (!(s->g))
```

Comment:
Please remove the redundant inner `()` here and at the other similar places below.